roles/create_day2_cluster/tasks/main.yml

```diff
@@ -48,3 +48,59 @@
   delegate_to: "{{ item }}"
   delegate_facts: yes
   loop: "{{ groups['day2_workers'] }}"
+
+#### patch discovery ignition on restricted network environments ###
+
+- name: Load patch for search registries
+  set_fact:
+    search_registries: "{{ lookup('template', 'patch-search-registries.j2') }}"
+  when: disconnected | bool == True
+
+- name: Load patch for discovery ignition
+  set_fact:
+    patch_discovery_ignition: "{{ lookup('template', 'patch-discovery-ignition.j2') }}"
+  when: disconnected | bool == True
+
+- name: Patch discovery ignition
+  uri:
+    #url: "{{ URL_ASSISTED_INSTALLER_CLUSTERS }}/{{ cluster_id }}/discovery-ignition"
+    url: "{{ ASSISTED_INSTALLER_BASE_URL }}/clusters/{{ add_host_cluster_id }}/discovery-ignition"
+    method: PATCH
+    status_code: [201]
+    return_content: True
+    body_format: json
+    body:
+      {
+        "config": "{{ patch_discovery_ignition | to_json(ensure_ascii=False) | string }}",
+      }
+  when: disconnected | bool == True
+  register: http_reply
+
+#### patch cluster install config on restricted network environments ###
+
+- name: Load patch for install config
```

Comment:
Do you need the install config as well? 
have you tried with just the discovery patches?
I haven't looked at the day2 process to in depth but I'm wondering it will pick that up as machine configs from the cluster.

Reply:
I'll need to revisit this as I can't recall and unable to test further at present. Should have a second shot in a few weeks.